tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -731,6 +735,62 @@ def convert_pytorch_to_onnx(model, input_shape, opset_version, example_inputs, o
     return model_onnx
 
 
+def convert_pytorch_model_pytorch_frontend(model, input_shape, example_inputs):
+    import torch
+    import inspect
+    try:
+        from openvino.frontend import FrontEndManager
+        from openvino.frontend.pytorch.decoder import TorchScriptPythonDecoder, pt_to_ov_type_map
+        fe_manager = FrontEndManager()
+        fe = fe_manager.load_by_framework('pytorch')
+    except Exception as e:
+        print("PyTorch frontend loading failed")
+        raise e
+    inputs = prepare_torch_inputs(example_inputs, input_shape)
+    model.eval()
+    input_signature = None
+    if isinstance(model, torch.nn.Module):
+        input_signature = list(inspect.signature(model.forward).parameters.keys())
+        try:
+            scripted = torch.jit.script(model)
+        except:
+            try:
+                scripted = torch.jit.trace(model, inputs)
+            except Exception as e:
+                print('Both traicing and scripting failed')
+                raise e
+    else:
+        scripted = model
+    f_model = torch.jit.freeze(scripted)
+    decoder = TorchScriptPythonDecoder(f_model)
+    im = fe.load(decoder)
+    om = fe.convert(im)
+    provide_shapes = input_shape is not None
+    inputs = [inputs] if isinstance(inputs, torch.Tensor) else inputs
+    if input_signature is not None and isinstance(inputs, dict):
+        ordered_inputs = []
+        for key in input_signature:
+            if key not in inputs:
+                continue
+            ordered_inputs.append(inputs[key])
+        inputs = ordered_inputs
+    for idx, input_tensor in enumerate(om.inputs):
+        if isinstance(inputs, (list, tuple)):
+            input_data = inputs[idx]
+        else:
+            input_data = list(inputs.values)[idx]
+        pt_dtype = input_data.dtype if isinstance(input_data, torch.Tensor) else type(input_data)
+        dtype = pt_to_ov_type_map.get(str(pt_dtype))
+        if dtype is None:
+            raise f"Unknown input dtype {pt_dtype}"
+
+        input_tensor.get_node().set_element_type(dtype)
+        shape = [-1] * len(input_data.shape) if not provide_shapes else input_shape[idx]
+        input_tensor.get_node().set_partial_shape(PartialShape(shape))
+    om.validate_nodes_and_infer_types()
+    return om
```

Comment:
I think we need to re-use FE configuration (loading, setting inputs and outputs) and rely on the existing implementation from https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/moc_frontend/pipeline.py

Reply:
reused